querygen/querygen.go

```diff
@@ -57,3 +57,26 @@ func GenBuildPipelineRunStartedQuery(index string) string {
 	)
 	return q
 }
+
+// GenBuildPipelineRunCompletedQuery returns a Splunk query for generating Segment events
+// representing success or failure of AppStudio build PipelineRuns.
+func GenBuildPipelineRunCompletedQuery(index string) string {
+	q, _ := UserJourneyQueryGen(
+		index,
+		`verb=update `+
+		`"responseStatus.code"=200 `+
+		`"objectRef.apiGroup"="tekton.dev" `+
+		`"objectRef.resource"="pipelineruns" `+
+		`"objectRef.subresource"="status" `+
+		`"responseObject.metadata.labels.pipelines.appstudio.openshift.io/type"=build `+
+		`"responseObject.metadata.resourceVersion"="*" `+
+		`"responseObject.status.startTime"="*" `+
+		`| spath responseObject.status.conditions{}.reason output=reason` +
```

Comment:
We should be able to remove this `spath` command by using `responseObject.status.conditions{}.reason` instead of `reason`.

Reply:
To do this I think you'll need to "dedupe" the value like the FieldSet code does. E.g.
```
eval event="Build PipelineRun ".mvindex('responseObject.status.conditions{}.reason', 0)
```